### PR TITLE
WT-5159 Fixes to allow SWIG4 to generate code compatible with Python3 or Python2.

### PR DESCRIPTION
**Pull request by ddanderson**
Tested with SWIG4 running with Python3 'in the tree'.  That is `cd build_posix; python3 ../test/suite/run.py ...` . Also tested it running with python2 'in the tree'.  Also did pip_setup.py (using python3, in case it matters), and was able to pip install/test the result in a python3 virtual env, as well as a python2 virtual env.

I made some of these tests with SWIG3.  That is, I can run python3 or python2 'in the tree'.  (These will be confirmed by the PR testing).  I did not check running pip_setup.py and pip install with SWIG3 .  The test matrix is getting too big.

If this works and passes review, I think we should merge, then upgrade all machines to SWIG4, and then deprecate SWIG3 and finally deprecate Python2.  We can deprecate in either order, but I think the SWIG3 deprecate will be easier (a single line in `build_posix/configure.ac.in`).

### CODE REVIEW

**Comment by ddanderson** [comment]
@lukech can you review please?

There's one thing that is confusing about this change and that is file naming.  Hopefully that will be less confusing once this is in.

The current state of affairs (before this change).  When doing a normal install into build_posix,
the SWIG-generated `wiredtiger.py` was installed as `build_posix/lang/python/wiredtiger/__init__.py`.  That's it!  When setting up a pip install, that didn't quite work, so we we put the generated `wiredtiger.py` into the installed directory (as `wiredtiger.py`) and had a special `pip_init.py` installed as `__init__.py`.  That special init imported the `wiredtiger.py`, and that allowed everything to work.

Turns out that SWIG4 changed how it did imports, which caused our build_posix installs to stop working.  And using the pip_init.py trick allows it to work in the tree. All except that the generated SWIG file cannot be named `wiredtiger.py`, it needs to be named something other than the package name (`wiredtiger`).

So, here's what it looks like after this change.  It's pretty much the same whether we're installing into the tree or doing a pip install.   The SWIG-generated `wiredtiger.py` is now copied to `build_posix/lang/python/wiredtiger/swig_wiredtiger.py` or for pip install as simply `swig_wiredtiger.py` in the installed directory.  And then special `init.py` (which used to be called `pip_init.py`) is installed as `build_posix/lang/python/wiredtiger/__init__.py` or for pip install as simply `__init__.py`.

I hope the changes make some sense with that in mind.

File: bench/workgen/runner/runner/__init__.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 # OTHER DEALINGS IN THE SOFTWARE.
 #
 # runner/__init__.py
-#	Used as a first import by runners, does any common initialization.
+#   Used as a first import by runners, does any common initialization.
```
Comment (ddanderson) [inline bench/workgen/runner/runner/__init__.py]: I needed to remove this tab because __init__.py is no longer a generated file anywhere in the tree.

**Comment by lukech** [comment]
@ddanderson Thanks much for making the changes and executing so many tests! I need some time to digest the change. 

I'm a bit curious on how to test SWIG4 on Linux distributions. With the new macOS version having SWIG4 as default version (via brew install), Ubuntu 18.04 has SWIG3 as default (via apt install). Do you need to manually compile SWIG4 on Linux? 

> When doing a normal install into build_posix
Do you mind sharing the detailed steps? I'm actually not sure how to efficiently test a swig related change. What I did before was git clean, reconfigure, and then make.

**Comment by lukech** [comment]
As the trigger of SWIG4 support was from the migration of macOS to 10.14, I kicked off an Evergreen patch build with `macos-1014` distro. Tasks turned green yay!

**Comment by ddanderson** [comment]
Yes, I built my own swig.  Here's roughly what I did, mostly from memory.  I did this on kodkod-aws, you are welcome to take or use my install directory:
 
```
$ mkdir $HOME/swig $HOME/swig/installed
$ cd $HOME/swig
(download swig tarball here)
$ tar -xzf swig-4.0.1.tar.gz
$ cd swig-4.0.1
(read INSTALL file)
$ configure --prefix=$HOME/swig/installed
$ make install
```
To use it, I set $SWIG in my environment to point to the installed swig: `SWIG=$(HOME)/swig/installed/bin/swig; export SWIG`
I believe the WT configure step will pick up the SWIG from the environment.  It's worth checking the config.log to make sure, it lists the SWIG version number.

Great to hear about macos 10.14!

**Pull request merged**